modality-probe-cli/src/export/graph.rs

```diff
@@ -0,0 +1,512 @@
+                        description: "four".to_string(),
+                        file: "four.c".to_string(),
+                        line: "4".to_string(),
+                    },
+                ),
+            ]
+            .into_iter()
+            .collect(),
+            events: vec![
+                (
+                    (a_uuid, 1),
+                    EventMeta {
+                        component_id: a_uuid,
+                        id: 1,
+                        name: "one".to_string(),
+                        type_hint: None,
+                        tags: String::new(),
+                        description: "one".to_string(),
+                        file: "one.c".to_string(),
+                        line: "1".to_string(),
+                    },
+                ),
+                (
+                    (a_uuid, 2),
+                    EventMeta {
+                        component_id: a_uuid,
+                        id: 2,
+                        name: "two".to_string(),
+                        type_hint: None,
+                        tags: String::new(),
+                        description: "two".to_string(),
+                        file: "two.c".to_string(),
+                        line: "2".to_string(),
+                    },
+                ),
+                (
+                    (a_uuid, 3),
+                    EventMeta {
+                        component_id: a_uuid,
+                        id: 3,
+                        name: "three".to_string(),
+                        type_hint: None,
+                        tags: String::new(),
+                        description: "three".to_string(),
+                        file: "three.c".to_string(),
+                        line: "3".to_string(),
+                    },
+                ),
+                (
+                    (a_uuid, 4),
+                    EventMeta {
+                        component_id: a_uuid,
+                        id: 4,
+                        name: "four".to_string(),
+                        type_hint: None,
+                        tags: String::new(),
+                        description: "four".to_string(),
+                        file: "four.c".to_string(),
+                        line: "4".to_string(),
+                    },
+                ),
+            ]
+            .into_iter()
+            .collect(),
+            probes_to_components: vec![(1, a_uuid), (2, a_uuid), (3, a_uuid), (4, a_uuid)]
+                .into_iter()
+                .collect(),
+        }
+    }
+
+    #[test]
+    fn complete_dot() {
+        let cfg = cfg();
+        let diamond_log = modality_probe_graph::test_support::diamond()
+            .into_iter()
+            .map(|e| (&e).try_into().unwrap())
+            .peekable();
+        let graph = super::log_to_graph(diamond_log).unwrap();
+
+        let dot = graph.graph.to_dot(&cfg).unwrap();
+        assert!(dot.contains("one_1_0 -> two_1_2"), dot);
```

Comment:
Is there an approach we could use for validating the output dot content's syntax/structure?

Reply:
Yeah, the thing you found below about the double `digraph` means this is obviously broken. I have not gotten to spend as much time testing here as I would like. Time willing I'd find a dot library and do some round-trips, I think.